Air Alarms are now visible in the dark

Poojawa opened the pull request:
:cl: Poojawa
add: Air Alarms are now visible in poor lighting conditions
/:cl:

I dunno about you, but seeing red APCs in the dark is pretty good to know. But if I'm peeking out into the darkness and trying to see if the hallway is vented, why...

![](https://i.gyazo.com/727e12f49277a98aef2e997ebdb76393.png)

That's not very helpful at all. But y'know what _would_ be?


![](https://i.gyazo.com/d853e42e60730e7cdd37c8252eecfeb4.gif)
![](https://i.gyazo.com/6aa6bbbc254b75c0af9f37294cf44ec1.gif)

Hopefully not...

![](https://i.gyazo.com/967bc601a8e3cb3fed8abdc2ebaa9564.gif)



But this is probably okay, enough.

![](https://i.gyazo.com/772fe6a9c61de76bfeda53ee359f0a10.gif)

I figured I'd spruce up Air alarms to use fancy `SSvis_overlays` like Computers and APCs do now. There's a very soft glow to each, despite the screenshots looking terribly dark. These are pretty coder sprite potato quality,  but usually people understand what the colors mean.


https://www.dropbox.com/s/3qegrmweks5sep6/airalarm2.mp4?dl=0 here's a video of how they look. Despite everything there's still a groddamn lack of seeing the glow, but I swears it's there. It's as powerful as the arcade machine's

Poojawa committed:
Aziz LIGHT

Cyberboss commented:
I like what you've done with this, but the highlight seems a little janky

Poojawa commented:
Indeed. The only one that came out decently was the red, the others are still a bit weird, but I'll try fluffing out the highlights a little

Poojawa commented:
![amber and red light](https://user-images.githubusercontent.com/6293118/45891662-c0f22a80-bd8b-11e8-8ab4-443556cea93e.gif)
![amber and red dark](https://user-images.githubusercontent.com/6293118/45891666-c2235780-bd8b-11e8-8c10-5e18ba183fc1.gif)
![green light](https://user-images.githubusercontent.com/6293118/45891667-c485b180-bd8b-11e8-9bab-0d5ff3b7e318.gif)
![green dark](https://user-images.githubusercontent.com/6293118/45891670-c5b6de80-bd8b-11e8-8d29-de1f1f18d8ed.gif)

How say you to these?

Poojawa committed:
What is divine light anyway

cleans up outlining. green is not a creative color.

Poojawa commented:
Updooted with revised and tweaked sprite states. Namely giving the 'off' tick a darker, non-highlight state so they're more legible in bright lighting.

Poojawa committed:
Why don't we take update_icon()

 and SHOVE IT OVER THERE

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -212,7 +217,11 @@
 /obj/machinery/airalarm/Initialize(mapload)
 	. = ..()
+	if(!mapload)
+		return
```
What's this block for? Why should this not run the other procs if instanced in regular play?

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -198,12 +204,11 @@
 		panel_open = TRUE
 		pixel_x = (dir & 3)? 0 : (dir == 4 ? -24 : 24)
 		pixel_y = (dir & 3)? (dir == 1 ? -24 : 24) : 0
+		update_icon()
```
You shouldn't need this. `update_icon()` is called in `power_change()` which is called during init. Just remove this from new.

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -198,12 +204,11 @@
+		update_icon()
 
 	if(name == initial(name))
 		name = "[get_area_name(src)] Air Alarm"
```
While you're dealing with the new proc here transfer the other easy to transfer things like this to init

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -212,7 +217,11 @@
 	. = ..()
+	if(!mapload)
+		return
+	power_change()
```
This is already called in `/obj/machinery/Initialize()`

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -212,7 +217,11 @@
+		return
+	power_change()
 	set_frequency(frequency)
+	update_icon()
```
This is already called in `power_change()`

Poojawa committed:
Air Alarms to Initialize fully

It can't be that easy... can it? 2 hours later edit: HAHAHAHAHAHA fuck

ChangelingRain reviewed (commented):
The letters probably shouldn't light up; it makes the sprite too busy.
The glow around the state indicator should be a rectangle, instead of missing corners.
`alarm_green` also shouldn't flash or have a ! on it, since it's the neutral state.

My feedback boils down to "these don't quite fit the sprite and are slightly lower-quality than the current air alarms"

ShizCalev commented:
#40048 +1

Poojawa commented:
Like the console screens, I don't really know how to fix FoW stuff. I'm guessing if you put it on top of the lighting plane it borks *all* of it up.

ninjanomnom commented:
It isn't fixable at all for us really without doing some really hacky shit. It needs engine features to handle properly. This is why I stopped making more of these.

Iamgoofball commented:
@Poojawa get yourself shareX and use webm

Cyberboss commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -826,8 +856,12 @@
-	update_icon()
+	if(circuit && !(flags_1 & NODECONSTRUCT_1)) //no circuit, no breaking
+		if(!(stat & BROKEN))
+			playsound(loc, 'sound/effects/glassbr3.ogg', 100, 1)
```
condense the 2 ifs and change loc to src

Poojawa committed:
if it ain't broke... break it.

Cyberboss commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -826,8 +856,11 @@
-	..()
-	update_icon()
+	if(circuit && !(flags_1 & NODECONSTRUCT_1) && !(stat & BROKEN))
+		playsound(src, 'sound/effects/glassbr3.ogg', 100, 1)
```
Why'd you remove the ..() call?

Poojawa replied:
Oh, that might actually be a me-goof, lemmie just revert it

Poojawa committed:
unless you break it, then you have to fix it.

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -816,7 +841,12 @@
+	else
+		set_light(brightness_on)
 	update_icon()
+	return
```
This return is unnecessary

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -816,7 +841,12 @@
+	if(stat & NOPOWER)
+		set_light(0)
+	else
+		set_light(brightness_on)
```
update_icon() called just below also sets the light, you have no reason to do it here

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -629,22 +645,29 @@
 /obj/machinery/airalarm/process()
 	if((stat & (NOPOWER|BROKEN)) || shorted)
-		return
+		return FALSE
```
process doesn't have a return value it cares about except `PROCESS_KILL` so there's no reason to add one here

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -681,6 +704,8 @@
 		mode = AALARM_MODE_SCRUBBING
 		apply_mode()
 
+	return TRUE
```
^

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -202,17 +209,17 @@
-	. = ..()
-	set_frequency(frequency)
+/obj/machinery/airalarm/on_construction()
+	..(dir,dir)
```
`/machinery` doesn't have any code to run for this proc and doesn't take any args. What are you doing here?

Poojawa committed:
cleans up embarressing shite code.

ninjanomnom commented on code/modules/atmospherics/machinery/airalarm.dm:
```diff
@@ -681,6 +701,8 @@
 		apply_mode()
 
+	return
+
```
You dont need returns at the end of procs like this

ninjanomnom commented:
As I said to oranges earlier I stopped making more things overlaid on top of the lighting layer because it has unsolvable issues without changes in byond. I wouldn't mind it much but honestly we probably shouldn't be using this until we can make sure it doesn't overlay on top of things like megafauna.

Poojawa committed:
Update airalarm.dm

Poojawa commented:
I'd say that's really just a minor cosmetic issue. There's just really no other practical means other than making the air alarms super bright - which just would be ugly, imo.

Poojawa committed:
Readding overlays to avoid vis_overlay complaints

People not seeing the vis_overylays would just see a white air alarm.
so, fixed

ninjanomnom commented:
![image](https://user-images.githubusercontent.com/1234602/46252966-25be1c80-c43f-11e8-810f-16b8d579aff5.png)
The issue isnt with vis overlays, it's with the feature concept as a whole. How does adding extra overlays fix anything about that?

Poojawa commented:
If you checked the comment, I explained why. I've noticed that sometimes the client doesn't have the vis_overlays actually overlaid, usually at roundstart upon instantly joining on box station etc, so air alarms would appear as just blank units. now they've got an overlay that'll show status, if not glow in the dark.

ninjanomnom commented:
Don't bandaid fix that issue, it's a byond problem which needs to get fixed

ninjanomnom commented:
You can't fix it, it's a byond problem

Poojawa commented:
So should I close this or...?

optimumtact commented:
stop using his bogus ass vis overlay thing for one then maybe it would work

ninjanomnom commented:
Regular overlays have the same issue

optimumtact commented:
for now lets not do this for anything else other than computer screens

Poojawa commented:
I mean, I guess? It's been merged on Citadel for a few weeks now with no ill issues.

optimumtact commented:
good for them, but this has layering issues with stuff appearing over the FOV layers
